File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    if speed = 0 then 0.0
    elif speed >= 1 && speed <= 4 then 1.0
    elif speed >= 5 && speed <= 8 then 0.9
    elif speed = 9 then 0.8
    else 0.77

let productionRatePerHour (speed: int) : float =
    float (speed * 221) * (successRate speed)

let workingItemsPerMinute (speed: int) : int =
    (productionRatePerHour speed) / 60.0 |> int

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    private static final int CARS_PER_HOUR = 221;

    public double productionRatePerHour(int speed) {
        return speed * CARS_PER_HOUR * successRate(speed);
    }

    private double successRate(int speed) {
        if (speed == 10) {
            return .77;
        }
        if (speed == 9) {
            return .8;
        }
        if (speed >= 5) {
            return .9;
        }
        return 1;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) productionRatePerHour(speed) / 60;
    }
}
